Program.cs

```diff
@@ -1,4 +1,91 @@
+// Copyright (c) Metamation India.
+// ------------------------------------------------------------------------
+// Program.cs
+// QUEUE<T>
+// Implement a Queue<T> using arrays as the underlying data structure. The queue should grow the array when needed (like the TStack above does).
+// If the array does not have to be grown, both Enqueue and Dequeue should be constant time (O(1)) operations. Throw exceptions as needed.
+// class TQueue<T> { 
+// public void Enqueue (T a) { }
+// public T Dequeue () { }
+// public T Peek () { }
+// public bool IsEmpty { get; }
+// }
+// --------------------------------------------------------------------------------------------
+using System;
 
-   
\ No newline at end of file
+namespace Training {
+   #region Program ------------------------------------------------------------------------------
+   class Program {
+      #region Method ---------------------------------------------
+      static void Main () {
+         TQueue<int> queue = new ();
+         queue.Enqueue (1);
+         queue.Enqueue (2);
+         queue.Enqueue (3);
+         Console.WriteLine ("Top item: " + queue.Peek ());
+         while (!queue.IsEmpty)
+            Console.WriteLine ("Dequeue item: " + queue.Dequeue ());
+      }
+      #endregion
+   }
+   #endregion
+
+   #region TQueue<T> ------------------------------------------------------------------------------
+   /// <summary>Create a class</summary>
+   /// <typeparam name="T">Datatype of the array</typeparam>
+   class TQueue<T> {
+      #region Constructor -------------------------------------------
+      /// <summary>Creating the constructor for initializing the values</summary>
+      public TQueue () {
+      }
+      #endregion
+
+      #region Implementation ----------------------------------------
+      /// <summary>Add a element to the array</summary>
+      /// <param name="a">Element to be add</param>
+      public void Enqueue (T a) {
+         if (mCount == mArray.Length) Array.Resize (ref mArray, mArray.Length * 2);
+         mArray[mRear] = a;
+         mRear = (mRear + 1) % mArray.Length;
+         mCount++;
+      }
+
+      /// <summary>Delete a element from the array and return it</summary>
+      /// <returns>Element to be return</returns>
+      public T Dequeue () {
+         InvalidException ();
+         T item = mArray[mFront];
+         mFront = (mFront + 1) % mArray.Length;
+         mCount--;
+         return item;
+      }
+
+      /// <summary>Display the top elememt</summary>
+      /// <returns>Top element in the array</returns>
+      public T Peek () {
+         InvalidException ();
+         return mArray[mFront];
+      }
+
+      /// <summary>Check the array is empty or not</summary>
+      public bool IsEmpty => mCount == 0;
+
+      /// <summary>Expection handling</summary>
+      /// <exception cref="InvalidOperationException">When the array is empty</exception>
+      public void InvalidException () {

```

Comment:
Have a proper function name. Function names should be **verb** or **verb + noun**.